src/util/tr.js

```diff
@@ -815,22 +912,44 @@ const _tr = (msg, {
     }
     iframe.setAttribute('style', 'z-index: 2147483647; display:block; height: 100% !important; width: 100% !important;min-height:80px !important;border:0px !important; border-color:none !important;')
     msgDiv.appendChild(iframe)
-    const ifrm = (iframe.contentWindow) ? iframe.contentWindow : (iframe.contentDocument.document) ? iframe.contentDocument.document : iframe.contentDocument
-    const doc = ifrm.document
 
     // Dispatch event for interstitial/exit intent close
     const closeCampaign = new Event('CT_campaign_rendered')
     document.dispatchEvent(closeCampaign)
 
-    doc.open()
-    doc.write(html)
     if (targetingMsgJson.display['custom-editor']) {
-      appendScriptForCustomEvent(targetingMsgJson, doc)
+      html = appendScriptForCustomEvent(targetingMsgJson, html)
+    }
+    iframe.srcdoc = html
+
+    let contentDiv
+    iframe.onload = () => {
+      if (targetingMsgJson.display['custom-editor'] && targetingMsgJson.display['custom-html-sandbox']) {
+        window.addEventListener('message', event => {
+          if (event?.data?.action === 'getnotifData') {
+            window.clevertap.renderNotificationClicked(event.data.value)
+          }
+          if (event?.data?.action === 'Event') {
+            window.clevertap.event.push(event.data.value)
+          } else if (event?.data?.action === 'Profile') {
+            window.clevertap.profile.push(event.data.value)
+          } else if (event?.data?.action === 'OUL') {
+            window.clevertap.onUserLogin.push(event.data.value)
+          } else if (event?.data?.action === 'closeInterstitialPopUp') {
+            var interstitialWrapper = window.document.getElementById('intentPreview')
+            var interstitialOverlay = window.document.getElementById('intentOpacityDiv')
+            setTimeout(() => {
```

Comment:
same comments as above. hard coded strings and switch statements

Reply:
Done!
Pushed the updated code!